src/plugins/workspace/public/services/workspace_validation_service.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { combineLatest, Subscription } from 'rxjs';
+import { i18n } from '@osd/i18n';
+
+import { recentWorkspaceManager } from '../recent_workspace_manager';
+import { WORKSPACE_DETAIL_APP_ID, WORKSPACE_FATAL_ERROR_APP_ID } from '../../common/constants';
+import {
+  ApplicationStart,
+  ChromeStart,
+  CoreSetup,
+  CoreStart,
+  WorkspaceError,
+} from '../../../../core/public';
+import { WorkspaceClient } from '../workspace_client';
+
+export class WorkspaceValidationService {
+  private workspaceClient?: WorkspaceClient;
+  private workspaceId: string | undefined;
```

Comment:
This property seems useless.

Reply:
I think we still need this as we need `recentWorkspaceManager.addRecentWorkspace(this.workspaceId!);`.